rinja_derive/src/generator.rs

```diff
@@ -2347,6 +2210,126 @@ impl<'a> Generator<'a> {
+        }) => {
+            if content.find('\\').is_none() {
+                // if the literal does not contain any backslashes, then it does not need unescaping
+                Cow::Borrowed(content)
+            } else {
+                // the input could be string escaped if it contains any backslashes
+                let input = format!(r#"'{content}'"#);
+                let input = input.parse().ok()?;
+                let input = syn::parse2::<syn::LitChar>(input).ok()?;
+                Cow::Owned(input.value().to_string())
+            }
+        }
+        Expr::NumLit(_, value) => {
+            enum NumKind {
+                Int(Option<IntKind>),
+                Float(Option<FloatKind>),
+            }
+
+            let (orig_value, kind) = match value {
+                Num::Int(value, kind) => (value, NumKind::Int(kind)),
+                Num::Float(value, kind) => (value, NumKind::Float(kind)),
+            };
+            let value = match orig_value.chars().any(|c| c == '_') {
+                true => Cow::Owned(orig_value.chars().filter(|&c| c != '_').collect()),
+                false => Cow::Borrowed(orig_value),
+            };
+
+            fn int<T: ToString, E>(
+                from_str_radix: impl Fn(&str, u32) -> Result<T, E>,
+                value: &str,
+            ) -> Option<String> {
+                Some(from_str_radix(value, 10).ok()?.to_string())
+            }
+
+            let value = match kind {
+                NumKind::Int(Some(IntKind::I8)) => int(i8::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::I16)) => int(i16::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::I32)) => int(i32::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::I64 | IntKind::Isize)) => {
+                    int(i64::from_str_radix, &value)?
+                }
+                NumKind::Int(Some(IntKind::I128)) => int(i128::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::U8)) => int(u8::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::U16)) => int(u16::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::U32)) => int(u32::from_str_radix, &value)?,
+                NumKind::Int(Some(IntKind::U64 | IntKind::Usize)) => {
+                    int(u64::from_str_radix, &value)?
+                }
+                NumKind::Int(Some(IntKind::U128)) => int(u128::from_str_radix, &value)?,
+                NumKind::Int(None) => match value.starts_with('-') {
+                    true => int(i128::from_str_radix, &value)?,
+                    false => int(u128::from_str_radix, &value)?,
+                },
+                NumKind::Float(Some(FloatKind::F32)) => value.parse::<f32>().ok()?.to_string(),
+                NumKind::Float(Some(FloatKind::F64) | None) => {
+                    value.parse::<f64>().ok()?.to_string()
+                }
+                // implement once `f16` and `f128` are available
```

Comment:
Please add a `FIXME` here and open an issue so it's not forgotten.

Reply:
Added.